torch/_inductor/codegen/common.py

```diff
@@ -557,23 +557,23 @@ def cpp_argdefs(self):
                 continue
             outer = inplaced.other_names[-1]
             inner = inplaced.inner_name
-            dtype = buffer_types[outer]
+            dtype = buffer_types.get(outer, V.graph.get_dtype(outer))
```

Comment:
We might be able to just call `V.graph.get_dtype(outer)` directly and delete `buffer_types` from this function.  I think `get_dtype` already handles constants (or we could easily make it handle them).